### PR TITLE
fix: Update override function calls reference to v6

### PR DESCRIPTION
**Pull request by nadetastic**
#### Description of changes

The override function calls for Authenticator reference APIs as they appear in v5 of `aws-amplify`. This PR updates them to reference usage in v6.
- https://ui.docs.amplify.aws/react/connected-components/authenticator/customization#override-function-calls

#### Issue #, if available
N/A

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/connected-components/authenticator/customization/customization.override-function-calls.web.mdx [hunk 1]
```diff
@@ -31,32 +31,32 @@ Each `handle*` function will return the neccessary values you'll need to make th
...
     </TableRow>
     <TableRow>
-      <ResponsiveTableCell label="Function Call">`Auth.confirmSignUp`</ResponsiveTableCell>
+      <ResponsiveTableCell label="Function Call">`confirmSignUp()`</ResponsiveTableCell>
       <ResponsiveTableCell label="Override Name">`handleConfirmSignUp`</ResponsiveTableCell>
       <ResponsiveTableCell label="formData Properties">`{username, code}`</ResponsiveTableCell>
     </TableRow>
     <TableRow>
-      <ResponsiveTableCell label="Function Call">`Auth.forgotPassword`</ResponsiveTableCell>
+      <ResponsiveTableCell label="Function Call">`resettPassword()`</ResponsiveTableCell>
```
Comment (esauerbo) [inline docs/src/pages/[platform]/connected-components/authenticator/customization/customization.override-function-calls.web.mdx:54]:
```suggestion
      <ResponsiveTableCell label="Function Call">`resetPassword()`</ResponsiveTableCell>
```

**Comment by esauerbo** [comment]
Thanks for doing this! One suggestion but otherwise LGTM

**Review by esauerbo — DISMISSED** [review]

**Review by reesscot — CHANGES_REQUESTED** [review]
We shouldn't fully remove this from the UI docs since most customers will still be using v5. We should show both versions on the page in a Tabs component.

File: docs/src/pages/[platform]/connected-components/authenticator/customization/customization.override-function-calls.react-native.mdx [hunk 2]
```diff
@@ -31,32 +31,32 @@ Each `handle*` function will return the neccessary values you'll need to make th
   </TableHead>
   <TableBody>
     <TableRow>
-      <ResponsiveTableCell label="Function Call">`Auth.signUp`</ResponsiveTableCell>
+      <ResponsiveTableCell label="Function Call">`signUp()`</ResponsiveTableCell>
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/customization/customization.override-function-calls.react-native.mdx:34]: Let's remove the parans on the function names, that is not a pattern that has been used in the UI docs previously, nor does it provide the correct input for the functions

**Review by calebpollman — CHANGES_REQUESTED** [review]

File: docs/src/pages/[platform]/connected-components/authenticator/customization/customization.override-function-calls.react-native.mdx [hunk 3]
```diff
@@ -31,32 +31,32 @@ Each `handle*` function will return the neccessary values you'll need to make th
   </TableHead>
   <TableBody>
     <TableRow>
-      <ResponsiveTableCell label="Function Call">`Auth.signUp`</ResponsiveTableCell>
```
[the remark this answers is not in the document]
Comment (nadetastic) [reply docs/src/pages/[platform]/connected-components/authenticator/customization/customization.override-function-calls.react-native.mdx:34]: Pushed a commit that adds a v5 column @reesscot how does this look?

File: docs/src/pages/[platform]/connected-components/authenticator/customization/overrides/username.react.mdx [hunk 4]
```diff
@@ -12,13 +13,15 @@ export default function AuthenticatorWithEmail() {
       // custom username
       username = username.toLowerCase();
       attributes.email = attributes.email.toLowerCase();
```
Comment (isaaccarrington) [inline docs/src/pages/[platform]/connected-components/authenticator/customization/overrides/username.react.mdx:15]: Can you add amend this to `options.userAttributes.email = options.userAttributes.email.toLowerCase();` please?

**Comment by esauerbo** [comment]
We updated the tables to reference v6 in https://github.com/aws-amplify/amplify-ui/pull/5153 and https://github.com/aws-amplify/amplify-ui/pull/5150, closing this out

**Pull request closed without merging**